### DISCUSSION TITLE
Sqitch with Yugabyte

### DISCUSSION THREAD

**Discussion by acamaral** [theory/sqitch#627] [Q&A] [answered]
Hi everybody,

I'm trying to use Saitch with **Yugabyte**, but on the first deploy I received the following error:

```
Adding registry tables to dev
ERROR:  EXCLUSIVE not supported yet
LINE 1: LOCK TABLE changes IN EXCLUSIVE MODE
```

Could someone help me with this issue? Any help will be wellcome.

Thanks in advance.

**Comment by theory** [comment]
Yugabyte is a new one to me. Sqitch does not have support for it, but is it supposed to be PostgreSQL-compliant? If so we might be able to fiddle with the registry schema similar to how it's done [for Postgres XC](/sqitchers/sqitch/blob/2dea22d/lib/App/Sqitch/Engine/pg.pm#L213-L219). In the meantime, you can try directly editing [Pg.sql](/sqitchers/sqitch/blob/develop/lib/App/Sqitch/Engine/pg.sql) to work around such errors.

**Reply by acamaral** [reply]
**Thank you** for the **quick** answer, David.

Yes, **Yugabyte** is a distributed  SQL database that is PostgreSQL-compliant, but there are some features that wasn't implemented yet, and  "_EXCLUSIVE_" is one of them.

I found out that the **sqitch** schema is created, so I don't think it's necessary to edit [pg.sql](https://github.com/sqitchers/sqitch/blob/develop/lib/App/Sqitch/Engine/pg.sql). The problem happens in [pg.pm](https://github.com/sqitchers/sqitch/blob/develop/lib/App/Sqitch/Engine/pg.pm) (line 262), when the command `LOCK TABLE changes IN EXCLUSIVE MODE` is sent to the database.

Anyway, I'll wait for the _EXCLUSIVE_ implementation by Yugabyte guys.

Thanks again for your attention.

**Comment by theory** [comment, marked as answer]
Right, of course, misplaced my noggin on what supports what. I've been super interested in Cockroach for a while, and now have a project where I need a good distributed database, so currently going down a rabbit hole as [Yugabyte compares itself to Cockroach](https://www.yugabyte.com/yugabytedb-vs-cockroachdb/) and [Cockroach responds](https://www.cockroachlabs.com/blog/unpacking-competitive-benchmarks/). I expect to add explicit support to Sqitch for one or the other before long. Appreciate the heads-up, super interesting!

**Comment by theory** [comment]
#635 adds support for Yugabyte. Work in progress, mainly to get the Docker image working in GitHub actions, and likely will tweak the detection of Yugabyte, but otherwise should work well enough. Note that explicit locking does not seem to be supported, so users must manually ensure that only one instances of Sqitch is modifying a database at a time. Might be able to correct that at some point, too; will ask around the Yuga community.
